akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -86,6 +82,53 @@ import akka.util.Helpers.toRootLowerCase
     (bytes(0) == GZIPInputStream.GZIP_MAGIC.toByte) &&
     (bytes(1) == (GZIPInputStream.GZIP_MAGIC >> 8).toByte)
   }
+
+  case class LZ4Meta(offset: Int, length: Int) {
+    import LZ4Meta._
+
+    def putInto(buffer: ByteBuffer): Unit = {
+      buffer.putInt(LZ4_MAGIC)
+      buffer.putInt(length)
+    }
+
+    def prependTo(bytes: Array[Byte]): Array[Byte] = {
+      val buffer = ByteBuffer.allocate(bytes.length + offset)
+      putInto(buffer)
+      buffer.put(bytes)
+      buffer.array()
+    }
+
+  }
+
+  object LZ4Meta {
+    val LZ4_MAGIC = 0x87d96df6 // The last 4 bytes of `printf akka | sha512sum`
+
+    def apply(bytes: Array[Byte]): LZ4Meta = {
+      LZ4Meta(8, bytes.length)
+    }
+
+    def get(buffer: ByteBuffer): OptionVal[LZ4Meta] = {
+      try {
+        if (buffer.getInt() != LZ4_MAGIC) {
+          OptionVal.None
+        } else {
+          OptionVal.Some(LZ4Meta(8, buffer.getInt()))
+        }
+      } catch {
+        case _: java.nio.BufferUnderflowException => OptionVal.None
+      }
+    }
+
+    def get(bytes: Array[Byte]): OptionVal[LZ4Meta] = {
+      get(ByteBuffer.wrap(bytes))
+    }
+
+  }
+
+  def isLZ4(bytes: Array[Byte]): Boolean = {
+    LZ4Meta.get(bytes).isDefined
```

Comment:
This will allocate a LZ4Meta for each LZ4-compressed message just to check and then throw away, could perhaps be worth optimizing away.

Edit: it is only used in tests, so skip having it in the production code at all instead?

Reply:
We already have `isGzipped`. I kept a isLZ4 just for symmetry. We usually do not call `isLZ4` directly. I don't think it is worthwhile to optimize this function.